bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala

```diff
@@ -496,6 +496,24 @@ case class TimeProblems(p: Proposal, s: ShellAdvisor) {
     case _ => None
   }
   lazy val b3ReqOrZero = b3Req.map(_.time).getOrElse(TimeAmount.empty)
+  lazy val jointNotAllowed = {
+    def checkForNotAllowedJointProposals(subs: Option[List[NgoSubmission]]):List[Option[Problem]] = {
+      val r = for {
+          subs <- subs
+          if subs.size > 1
+        } yield for {
+            p <- subs.filter(s => Partners.jointProposalNotAllowed.contains(s.partner))
+          } yield new Problem(Severity.Error, s"${~Partners.name.get(p.partner)} cannot be part of a joint proposal, please update the time request.", SCHEDULING_SECTION,
+              s.inPartnersView(_.editSubmissionTime(p)))
+      r.sequence
+    }
+
+    p.proposalClass match {
+      case p: GeminiNormalProposalClass => checkForNotAllowedJointProposals(p.subs.left.toOption)
+      case e: ExchangeProposalClass     => checkForNotAllowedJointProposals(e.subs.some)
+      case x                            => println("abc " + x);Nil
+    }
+  }
```

Comment:
Hi, right a List[Problem] will do and certainly debugging will be removed
Funny, I find clearer the nested `for` (no more than 2 levels)  than a combination of operations, should we have some guidelines about that?